kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/JSONInputFormat.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2013 Cloudera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.kitesdk.data.spi.filesystem;
+
+import java.io.IOException;
+import java.util.List;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.InputSplit;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.RecordReader;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+import org.kitesdk.compat.Hadoop;
+import org.kitesdk.data.DatasetDescriptor;
+import org.kitesdk.data.spi.DataModelUtil;
+import org.kitesdk.data.spi.EntityAccessor;
+
+class JSONInputFormat<E> extends FileInputFormat<E, Void> {
+  private DatasetDescriptor descriptor = null;
+  private EntityAccessor<E> accessor = null;
+
+  public void setDescriptor(DatasetDescriptor descriptor) {
+    this.descriptor = descriptor;
+  }
+
+  public void setType(Class<E> type) {
+    this.accessor = DataModelUtil.accessor(type, descriptor.getSchema());
+  }
+
+  @Override
+  protected boolean isSplitable(JobContext context, Path filename) {
+    return false;
```

Comment:
That's why I was saying it should be an option, and not splittable by default. If `}{` is not valid internal to JSON record, then we could use that to be even more robustness. This is even more apropos to my comment about documenting the file format(s) we expect/can handle.